### ISSUE TITLE
VPC Serverless Connector module is recreated every time

### ISSUE THREAD

**Issue by tim-codes** [terraform-google-modules/terraform-google-network#357]
### TL;DR

Some recent change (must be in the API) causes the serverless connector beta module to be recreated every time, due to network field being set to null. This is in the scenario where the module is supplied with subnet name (for shared VPC) rather than network and ip_cidr_range fields.

Redacted plan output:
```
resource "google_vpc_access_connector" "connector_beta" {
      ...
      - network        = "main" -> null # forces replacement
      ...
}
```

Network field cannot be supplied and I cannot set ignore changes through the module to workaround this.

I tried reverting module versions but this didn't help and this issue only came up this week so can't be a problem with the module.

### Terraform Configuration

```hcl
// using TS CDK
const vpcServerlessConnectors = new TerraformHclModule(
    stack,
    stack.toUid('vpc-serverless-connector'),
    {
      source: 'terraform-google-modules/network/google//modules/vpc-serverless-connector-beta',
      variables: {
        project_id: stack.project.id,
        vpc_connectors: operatingRegions
          .map((r) => [r, regions[r]] as [ToolingRegions, Region])
          .map(([r, region]) => ({
            name: `connector-${r}`,
            region: region.id,
            host_project_id: stack.project.id, // Specify a host_project_id for shared VPC
            subnet_name: subnets.gcpServerless[r].name,
            machine_type: 'e2-micro',
            min_instances: 2,
            max_instances: 4,
            // specify in order to stop this value being replaced on each run
            max_throughput: 400,
          })),
      },
    },
  );
```

### Terraform Version

```sh
Terraform v1.1.7
on darwin_amd64
```

**Comment by morgante** [comment]
Please report changes in the API on the provider: https://github.com/hashicorp/terraform-provider-google
